Performance/perf_runner.sh

```diff
@@ -217,39 +274,95 @@ if [[ ! -f "$results_js" ]]; then
 fi
 
 # Iterate over the requested field types and run the harnesses.
+# TODO: Get rid of this multi-run; instead, just create a proto like
+# TestAllTypes that tests multiple field types in one message. This is more
+# realistic.
 for field_type in "${requested_field_types[@]}"; do
   gen_message_path="$script_dir/_generated/message.proto"
 
   echo "Generating test proto with $field_count fields of type $field_type..."
   generate_test_proto "$field_count" "$field_type"
 
-  protoc --plugin="$script_dir/../.build/release/protoc-gen-swiftForPerf" \
-      --swiftForPerf_out=FileNaming=DropPath:"$script_dir/_generated" \
-      --cpp_out="$script_dir" \
-      "$gen_message_path"
-
   # Start a session.
   partial_results="$script_dir/_results/partial.js"
+  pretty_head_rev="$(git rev-parse --abbrev-ref HEAD)"
   cat > "$partial_results" <<EOF
   {
     date: "$(date -u +"%FT%T.000Z")",
     type: "$field_count fields of type $field_type",
-    branch: "$(git rev-parse --abbrev-ref HEAD)",
+    branch: "$pretty_head_rev",
     commit: "$(git rev-parse HEAD)",
     uncommitted_changes: $([[ -z $(git status -s) ]] && echo false || echo true),
+    series: [
 EOF
 
-  harness_cpp="$script_dir/_generated/harness_cpp"
-  results_trace="$script_dir/_results/$field_count fields of $field_type (cpp)"
-  run_cpp_harness "$harness_cpp"
+  for comparison in "${comparisons[@]}"; do
+    if [[ "$comparison" == "c++" ]]; then
+      source "$script_dir/runners/cpp.sh"
+
+      echo
+      echo "==== Building/running C++ harness ===================="
+      echo
+
+      protoc --cpp_out="$script_dir" "$gen_message_path"
+
+      harness_cpp="$script_dir/_generated/harness_cpp"
+      results_trace="$script_dir/_results/$field_count fields of $field_type (cpp)"
+      run_cpp_harness "$harness_cpp"
+    else
+      commit_hash="$(git rev-parse $comparison)"
+      commit_results="$script_dir/_results/${commit_hash}_${field_type}_${field_count}.js"
+
+      # Check to see if we have past results from that commit cached. If so, we
+      # don't need to run it again.
+      if [[ ! -f "$commit_results" ]]; then
+        echo
+        echo "==== Building/running Swift harness ($comparison) ===================="
+        echo
+
+        # Check out the commit to a temporary directory and create its _generated
+        # directory. (Results will still go in the working tree.)
+        tmp_checkout="$(mktemp -d -t swiftprotoperf)"
```

Comment:
do you want an exit hook to delete this directory?

Early on, set up a hook:
```bash
  CLEANUP_WHEN_DONE=()
  function cleanup() {
    rm -rf "${CLEANUP_WHEN_DONE[@]}"
  }
  trap cleanup EXIT HUP INT QUIT TERM
```

The this directory creation becomes:

```bash
  tmp_checkout="$(mktemp -d -t swiftprotoperf)"
  CLEANUP_WHEN_DONE+=("$tmp_checkout")
```

Then no mater now the script dies/exits (unless it is an exec), it will ensure everything added to CLEANUP_WHEN_DONE gets removed at the end.